plugins/scaffolder/src/components/fields/RepoUrlPicker/BitbucketRepoPicker.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import FormControl from '@material-ui/core/FormControl';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import Input from '@material-ui/core/Input';
+import InputLabel from '@material-ui/core/InputLabel';
+
+export const BitbucketRepoPicker = ({
+  onProjectChange,
+  onWorkspaceChange,
+  onRepoNameChange,
+  rawErrors,
+  workspace,
+  project,
+  host,
+  repoName,
+}: {
+  onProjectChange: (owner: string) => void;
+  onWorkspaceChange: (name: string) => void;
+  onRepoNameChange: (name: string) => void;
```

Comment:
it'd be nice to us (but maybe less nice to a consumer?) to have a single onChange that is passed the full output object, i'd say

Reply:
and at the same time - it could be given that same type as a single input?

```ts
{
  onChange: (data: T) => void,
  data: T,
```

arguably makes for a more consistent and simple API